goal_src/engine/target/logic-target.gc

```diff
@@ -6,38 +6,3086 @@
+    (let*
+     ((a1-2
+       (vector-flatten!
+        (new 'stack-no-clear 'vector)
+        (vector-negate!
+         (new-stack-vector0)
+         (-> self control dynam gravity-normal)
+         )
+        (-> self control local-normal)
+        )
+       )
+      (s1-0
+       (vector-matrix*!
+        (new 'stack-no-clear 'vector)
+        a1-2
+        (-> self control unknown-matrix00)
+        )
+       )
+      (s3-1 (vector-normalize-copy! (new 'stack-no-clear 'vector) s1-0 1.0))
+      )
+     (vector+float*!
+      s5-0
+      s5-0
+      s1-0
+      (-> self control unknown-surface01 slide-factor)
+      )
+     (let ((s2-1 (-> self control unknown-vector60)))
+      (when
+       (not
+        (or
+         (= (- 1.0 (-> self control unknown-surface01 slope-up-traction)) 0.0)
+         (< (vector-length s1-0) 0.1)
+         )
+        )
+       (let*
+        ((s1-1
+          (vector-matrix*!
+           (new 'stack-no-clear 'vector)
+           s2-1
+           (-> self control unknown-matrix00)
+           )
+          )
+         (s2-2 (new-stack-vector0))
+         (f30-0 (vector-dot s1-1 s5-0))
+         )
+        0.0
+        (vector-! s2-2 s5-0 (vector-float*! s2-2 s1-1 f30-0))
+        (let* ((f0-7 (vector-length s2-2))
+               (f1-3 f0-7)
+               )
+         (if (< f30-0 0.0)
+          (set!
+           f30-0
+           (* f30-0 (-> self control unknown-surface01 slope-up-traction))
+           )
+          )
+         (let
+          ((f2-3
+            (+ f30-0 (-> self control unknown-surface01 slope-down-factor))
+            )
+           )
+          (vector+!
+           s5-0
+           (vector-float*! s5-0 s1-1 f2-3)
+           (vector-float*! s2-2 s2-2 (/ f0-7 f1-3))
+           )
+          )
+         )
+        )
+       )
+      )
+     (let ((t9-8 vector-xz-normalize!)
+           (a0-13 (new-stack-vector0))
+           )
+      (set! (-> a0-13 quad) (-> s5-0 quad))
+      (let ((v1-31 (t9-8 a0-13 1.0)))
+       (set! (-> v1-31 y) 0.0)
+       (let ((f30-1 (vector-dot s3-1 v1-31))
+             (f0-11 (vector-xz-length s5-0))
+             )
+        (if (< 0.0 f30-1)
+         (vector-xz-normalize!
+          s5-0
+          (+
+           f0-11
+           (* f30-1 (-> self control unknown-surface01 slope-down-factor))
+           )
+          )
+         (vector-xz-normalize!
+          s5-0
+          (-
+           f0-11
+           (*
+            (* f30-1 f30-1)
+            (-> self control unknown-surface01 slope-up-factor)
+            )
+           )
+          )
+         )
+        )
+       )
+      )
+     )
+    (let ((t9-12 vector-xz-normalize!)
+          (a0-17 (new-stack-vector0))
+          )
+     (set! (-> a0-17 quad) (-> s5-0 quad))
+     (let ((s3-2 (t9-12 a0-17 1.0)))
+      (let ((t9-13 vector-xz-normalize!)
+            (a0-18 (new-stack-vector0))
+            )
+       (set! (-> a0-18 quad) (-> gp-0 quad))
```

Comment:
I've seen this pattern a lot, I think it's a macro to create a new stack vector and copy an existing.  I will add it to the decompiler at some point so this looks nicer.